deepspeed/runtime/engine.py

```diff
@@ -721,6 +721,9 @@ def _do_sanity_check(self):
                 assert self._is_supported_optimizer(self.optimizer_name()), \
                     '{} is not a supported DeepSpeed Optimizer'.format(self.optimizer_name())
 
+        assert self.client_optimizer is not None or self.client_model_parameters is not None, \
```

Comment:
1. Add assertion to DS engine. see https://github.com/microsoft/DeepSpeed/issues/1034